test/core/credentials/transport/tls/spiffe_utils_test.cc

```diff
@@ -34,106 +34,331 @@
-        {"TooLong", std::string(2049, 'a'),
-         "maximum allowed for SPIFFE ID is 2048"},
-        {"ContainsHashtag", "ab#de", "cannot contain query fragments"},
-        {"ContainsQuestionMark", "ab?de", "cannot contain query parameters"},
-
-        {"DoesNotStartWithSpiffe", "www://foo/bar",
-         "must start with spiffe://"},
-        {"EndsWithSlash", "spiffe://foo/bar/", "cannot end with a /"},
-        {"NoTrustDomain", "spiffe://", "cannot end with a /"},
-        {"NoTrustDomainWithPath", "spiffe:///path",
-         "The trust domain cannot be empty"},
-        {"TrustDomainTooLong", absl::StrCat("spiffe://", std::string(256, 'a')),
-         "Trust domain maximum length is 255 characters"},
-        {"TrustDomainInvalidCharacter1", "spiffe://bad@domain",
-         "contains invalid character @"},
-        {"TrustDomainInvalidCharacter2", "spiffe://BadDomain",
-         "contains invalid character B"},
-        {"PathContainsRelativeModifier1", "spiffe://example/path/./foo",
-         ". or .."},
-        {"PathContainsRelativeModifier2", "spiffe://example/path/../foo",
-         ". or .."},
-        {"PathSegmentBadCharacter", "spiffe://example/path/foo.bar/foo@bar",
-         "invalid character @"},
-        {"ContainsNonASCIITrustDomain", "spiffe://µ/path",
-         "cannot contain non-ascii characters"},
-        {"ContainsNonASCIIPath", "spiffe://foo.bar/µ",
-         "cannot contain non-ascii characters"},
-    }),
-    [](const ::testing::TestParamInfo<SpiffeIdFailureTest::ParamType>& info) {
-      return info.param.test_name;
-    });
-
-struct SpiffeIdSuccessTestCase {
-  std::string spiffe_id;
-  std::string trust_domain;
-  std::string path;
-};
-
-using SpiffeIdSuccessTest = TestWithParam<SpiffeIdSuccessTestCase>;
-
-TEST_P(SpiffeIdSuccessTest, SpiffeIdTestSuccess) {
-  const SpiffeIdSuccessTestCase& test_case = GetParam();
+              ::testing::HasSubstr("Path segment contains invalid character"));
+}
+
+TEST(SpiffeId, FragmentInPathFails) {
+  EXPECT_EQ(SpiffeId::FromString("spiffe://foo/pa^h").status(),
+            absl::InvalidArgumentError(
+                "Path segment contains invalid character ^. MUST contain only "
+                "letters, numbers, dots, dashes, and underscores"));
+}
+
+TEST(SpiffeId, MultipleSlashesInPathFails) {
+  EXPECT_EQ(SpiffeId::FromString("spiffe://foo/bar//baz").status(),
+            absl::InvalidArgumentError("Path segment cannot be empty"));
+}
+
+TEST(SpiffeId, NonASCIIPathFails) {
+  EXPECT_EQ(SpiffeId::FromString("spiffe://foo.bar/µ").status(),
+            absl::InvalidArgumentError(
+                "SPIFFE ID URI cannot contain non-ascii characters"));
+}
+
+TEST(SpiffeId, ContainsNonASCIIPathFails) {
+  EXPECT_EQ(SpiffeId::FromString("spiffe://foo.bar/fooµbar").status(),
+            absl::InvalidArgumentError(
+                "SPIFFE ID URI cannot contain non-ascii characters"));
+}
+
+TEST(SpiffeId, NoPathSuccess) {
+  absl::StatusOr<SpiffeId> spiffe_id =
```

Comment:
Consider using `auto` here instead of `absl::StatusOr<SpiffeId>` to make the tests a little shorter.